packages/scene-composer/src/components/modals/DeleteNodeModal.tsx

```diff
@@ -0,0 +1,117 @@
+  const { removeSceneNode } = useSceneDocument(sceneComposerId);
+
+  const [childCount, setChildCount] = useState<number | undefined>();
+  const [loadChildCountError, setLoadChildCountError] = useState<Error | undefined>();
+
+  const nodeRef = deleteConfirmationModalParams?.nodeRef;
+  const node = getSceneNodeByRef(nodeRef);
+  const nodeName = node?.name ?? '';
+
+  const title = formatMessage(
+    { defaultMessage: 'Delete {nodeName}', description: 'Delete node confirmation header text' },
+    { nodeName },
+  );
+
+  useEffect(() => {
+    const executeQuery = getGlobalSettings().twinMakerSceneMetadataModule?.kgModule.executeQuery;
+    if (executeQuery && nodeRef) {
+      const statement = `SELECT COUNT(e) FROM EntityGraph MATCH (e)-[:${DEFAULT_PARENT_RELATIONSHIP_NAME}]->{1, ${MAX_QUERY_HOP}}(p) WHERE p.entityId = '${nodeRef}'`;
+
+      executeQuery({ queryStatement: statement })
+        .then((response) => {
+          const count = response.rows?.at(0)?.rowData?.at(0);
+          if (count !== undefined && isNumber(count)) {
+            setChildCount(count);
+          } else {
+            setChildCount(-1);
+          }
+        })
+        .catch((error) => {
+          setLoadChildCountError(error);
+        });
+    }
+  }, []);
+
+  const onDelete = useCallback(() => {
+    if (nodeRef) {
+      removeSceneNode(nodeRef);
+    }
+    setDeleteConfirmationModalVisible(false);
+  }, [nodeRef, removeSceneNode]);
+
+  const contentMessage = useCallback(() => {
+    if (childCount === undefined && loadChildCountError === undefined) {
+      return <Spinner />;
+    } else if (childCount !== undefined && childCount >= 0) {
+      if (childCount === 0) {
+        return formatMessage(
+          {
+            defaultMessage:
+              'Delete {nodeName}? This will delete 1 entity. While this action may be undone, it may not return to its original state.',
+            description: 'Delete action confirmation message',
+          },
+          { nodeName },
+        );
+      } else {
+        return formatMessage(
+          {
+            defaultMessage:
+              'Delete {nodeName}? This will delete 1 entity and its {childCount} children. While this action may be undone, it may not return to its original state.',
+            description: 'Delete action confirmation message',
+          },
+          { childCount, nodeName },
+        );
+      }
+    } else {
+      return formatMessage(
+        {
+          defaultMessage:
+            'Delete {nodeName}? Unable to load number of children. While this action may be undone, it may not return to its original state.',
+          description: 'Delete action confirmation message',
+        },
+        { nodeName },
+      );
+    }
+  }, [childCount, loadChildCountError, formatMessage, nodeName]);
+
+  return (
+    <DeleteConfirmationModal
+      title={title}
+      contentBody={contentMessage()}
+      warningMessage={formatMessage(
```

Comment:
NIT: This can also be moved out as a `const` same as `title` to maintain consistency.

Reply:
I'll move in next change